tests/rptest/services/action_injector.py

```diff
@@ -33,6 +34,41 @@ def time_between_actions(self) -> float:
                               self.max_time_between_actions_sec)
 
 
+class GenerationPriority:
+    """
+    Calculates random priorities for items, grouping them by generations.
+    For the first max_items calls to next_priority, the returned priorities
+    are between 0 to step. For the next max_items calls, the priorities are
+    from step+1 to 2*step. For each generation the priorities are in a range
+    higher than previous and less than next generation, so that using the
+    returned priority in a queue ensures that items from one generation are
+    picked before items from the next generation.
+    """
+    def __init__(self, max_items: int, step: int = 1000):
+        self._step = step
+        self._max_items = max_items
+        self._generation = 0
+        self._items = 0
+        random.seed(time.time())
+
+    def next_priority(self) -> int:
+        self._items += 1
+        if self._items > self._max_items:
+            self._items = self._items % self._max_items
+            self._generation += 1
+        return random.randint(self._generation * self._step,
+                              (self._generation + 1) * self._step)
```

Comment:
Nit: looks like because for `randint` the boundaries are inclusive, we have a small chance of overlapping priorities for different generations here. So it is better to use `randrange` (and comment needs to be fixed a bit too).